### PR TITLE
DPE-70 Add CI 

### PR DESCRIPTION
**Pull request by WRFitch**
## Proposal
Currently the container for pgbouncer has no CI  to lint or build an image. This PR adds these features, as well as a licence, .gitignore file, and a pull request template. 

This CI can later be extended to include [DPE-70](https://warthogs.atlassian.net/browse/DPE-70?atlOrigin=eyJpIjoiZTM2NTU4ZGNjZjY4NGQxOTkyYWJjNWIyZjkzMjkzMmYiLCJwIjoiaiJ9), automatically deploying this container to dockerhub. 

## Release Notes
- Added linting CI
- Added build & deploy test CI
- Added License
- Added pull request template 

## Testing
These have been run on a [personal fork](https://github.com/WRFitch/pgbouncer-container/actions?query=branch%3ADPE-70-add-ci++).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +35,4 @@ Pgbouncer won't run without the following environment variables passed in:
   - These will be encrypted in future.
   - A default "pw" is provided if the default `$PGB_ADMIN_USERS` value is used, but otherwise no default is provided.
 
+    1. `docker run -rm --name pgbouncer:local`
```
Comment (marceloneppel) [inline README.md]: This addition needs some context/explanation. It's not clear why it's here.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Lint Dockerfile
+        uses: hadolint/hadolint-action@v1.6.0
+        with:
+          dockerfile: dockerfile
+
+      - name: Lint Entrypoint Script
```
Comment (marceloneppel) [inline .github/workflows/lint.yml:22]: Nice addition!

**Review by marceloneppel — APPROVED** [review]
LGTM! Do you know why the CI is not running in this PR?

File: dockerfile [hunk 3]
```diff
@@ -1,8 +1,10 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
```
Comment (marcoppenheimer) [inline dockerfile]: Not this line, but the file should be Capitalised `Dockerfile`

File: entrypoint.sh [hunk 4]
```diff
@@ -8,7 +8,7 @@ USERLIST="${PGB_DIR}/userlist.txt"
 rm -f "${INI}" "${USERLIST}"
```
Comment (marcoppenheimer) [inline entrypoint.sh:8]: Likewise here, I'd rename the file to `docker-entrypoint.sh`

**Review by marcoppenheimer — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (WRFitch) [reply README.md]: Not sure what this is, but it's been removed

**Comment by WRFitch** [comment]
> LGTM! Do you know why the CI is not running in this PR?

I assume this is because it's not merged into master yet, but you can find some existing test runs in [a separate PR I made on the fork](https://github.com/WRFitch/pgbouncer-container/pull/1)

**Pull request merged**